experimental/SetPartitions/src/ColoredPartition.jl

```diff
@@ -0,0 +1,164 @@
+import Base.hash
+import Base.==
+import Base.copy
+
+"""
+ColoredPartition
+
+Initialize colored Partition object
+
+# Arguments
+- `partition`: Partition object which is generalized with coloring `color_upper_points` and `color_lower_points`
+- `color_upper_points`: color (in [0, 1]) of upper points
+- `color_lower_points`: color (in [0, 1]) of lower points
+"""
+struct ColoredPartition <: AbstractPartition
+    partition::Partition
+    color_upper_points::Array{Int64, 1}
+    color_lower_points::Array{Int64, 1}
+end
+
+function colored_partition(partition::Partition, color_upper::Array{Int64, 1}, color_lower::Array{Int64, 1})
+
+    return ColoredPartition(partition, color_upper, color_lower)
+
+end
+
+function hash(p::ColoredPartition, h::UInt)
+
+    hash(p.partition, hash(p.color_upper_points, hash(p.color_lower_points, h)))
+    
+end
+
+function ==(p::ColoredPartition, q::ColoredPartition)
+
+    p.partition == q.partition && p.color_upper_points == q.color_upper_points && p.color_lower_points == q.color_lower_points
+
+end
+
+function copy(p::ColoredPartition)
+    return ColoredPartition(copy(p.partition), copy(p.color_upper_points), copy(p.color_lower_points))
+end
+
+
+"""
+tensor_product(p::ColoredPartition, q::ColoredPartition)
```

Comment:
```suggestion
    tensor_product(p::ColoredPartition, q::ColoredPartition)
```
These lines should be indented by four spaces, which in Markdown indicates a code block.